tests/galaxy/galaxy_fx.py

```diff
@@ -95,6 +95,12 @@ def mock_system_api_server_fx():
             """{"data":[]}"""
         )
 
+        # Tests that Galaxy will retry failed requests
+        httpserver.expect_oneshot_request("/badendpoint").respond_with_data(status = 400)
+        httpserver.expect_oneshot_request("/badendpoint").respond_with_data(status = 500)
+        httpserver.expect_oneshot_request("/badendpoint").respond_with_data("""{"success":true}""")
```

Comment:
[question] how does this work, having a single endpoint that has multiple returns?

Reply:
Each of those are defined as "oneshot" requests. Meaning that once `pytest_httpserver` has responded to it, it will remove that response from its list.

Since the test lib searches from the front to the back of the array, it will find the first oneshot request, use it to respond, then remove it. The next call will use the second response, and remove it. And so on and so forth.

Reply:
wierd but ok. thanks for explaining that.